lib/util.js

```diff
@@ -235,12 +235,13 @@
    * @param String aStr
    */
   function toSetString(aStr) {
-    return '$' + aStr;
+    if (/^\$*__proto__$/.test(aStr)) return '$' + aStr;
```

Comment:
Nitpick: match the existing style:

```js
if (condition) {
  then();
}
```

Did you see if pulling the regex definition out of the method helped performance at all? I don't know whether most JS engines would see through this and avoid allocating a `RegExp` object or not (I know spidermonkey will at least re-use and share the compiled regex at a source location between `RegExp` objects, but I don't know if it would avoid allocating the `RegExp` object all together in this particular case).

I suspect it is faster either way to test the string "by hand" rather than with a regular expression:

```js
function isProtoString(s) {
  var length = s.length;

  if (length < 9 /* "__proto__".length */) {
    return false;
  }

  if (s.charCodeAt(length - 1) !== 95  /* '_' */ ||
      s.charCodeAt(length - 2) !== 95  /* '_' */ ||
      s.charCodeAt(length - 3) !== 111 /* 'o' */ ||
      s.charCodeAt(length - 4) !== 116 /* 't' */ ||
      s.charCodeAt(length - 5) !== 111 /* 'o' */ ||
      s.charCodeAt(length - 6) !== 114 /* 'r' */ ||
      s.charCodeAt(length - 7) !== 112 /* 'p' */ ||
      s.charCodeAt(length - 8) !== 95  /* '_' */ ||
      s.charCodeAt(length - 9) !== 95  /* '_' */) {
    return false;
  }

  for (var i = length - 10; i >= 0; i--) {
    if (s.charCodeAt(i) !== 36 /* '$' */) {
      return false;
    }
  }

  return true;
}
```